x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyServiceTests.java

```diff
@@ -200,6 +212,130 @@ public void testCreateApiKeyUsesBulkIndexAction() throws Exception {
+            } else {
+                boolQuery.filter(QueryBuilders.termQuery("name", apiKeyName));
+            }
+        }
+        if (apiKeyIds != null && apiKeyIds.length > 0) {
+            boolQuery.filter(QueryBuilders.idsQuery().addIds(apiKeyIds));
+        }
+        verify(searchRequestBuilder).setQuery(eq(boolQuery));
+        verify(searchRequestBuilder).setFetchSource(eq(true));
+        assertThat(searchRequest.get().source().query(), is(boolQuery));
+        GetApiKeyResponse getApiKeyResponse = getApiKeyResponsePlainActionFuture.get();
+        assertThat(getApiKeyResponse.getApiKeyInfos(), emptyArray());
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testInvalidateApiKeys() throws Exception {
+        final Settings settings = Settings.builder().put(XPackSettings.API_KEY_SERVICE_ENABLED_SETTING.getKey(), true).build();
+        when(client.threadPool()).thenReturn(threadPool);
+        SearchRequestBuilder searchRequestBuilder = Mockito.spy(new SearchRequestBuilder(client, SearchAction.INSTANCE));
+        when(client.prepareSearch(eq(SECURITY_MAIN_ALIAS))).thenReturn(searchRequestBuilder);
+        final ApiKeyService service = createApiKeyService(settings);
+        final AtomicReference<SearchRequest> searchRequest = new AtomicReference<>();
+        doAnswer(invocationOnMock -> {
+            searchRequest.set((SearchRequest) invocationOnMock.getArguments()[0]);
+            ActionListener<SearchResponse> listener = (ActionListener<SearchResponse>) invocationOnMock.getArguments()[1];
+            listener.onResponse(SearchResponse.empty(() -> 1L, SearchResponse.Clusters.EMPTY));
+            return null;
+        }).when(client).search(any(SearchRequest.class), anyActionListener());
+        PlainActionFuture<InvalidateApiKeyResponse> listener = new PlainActionFuture<>();
+        service.invalidateApiKeys(
+            randomFrom(new String[0], null),
+            randomFrom("", null),
+            randomFrom("", null),
+            randomFrom(new String[0], null),
+            listener
+        );
+        ExecutionException e = expectThrows(ExecutionException.class, () -> listener.get());
+        assertThat(e.getCause(), instanceOf(IllegalArgumentException.class));
+        assertThat(e.getCause().getMessage(), containsString("One of [api key id, api key name, username, realm name] must be specified"));
+        String[] realmNames = generateRandomStringArray(4, 4, true, true);
+        String username = randomFrom(randomAlphaOfLengthBetween(3, 8), null);
+        String apiKeyName = randomFrom(randomAlphaOfLengthBetween(3, 8), null);
+        String[] apiKeyIds = generateRandomStringArray(4, 4, true, true);
+        if ((realmNames == null || realmNames.length == 0)
+            && Strings.hasText(username) == false
+            && Strings.hasText(apiKeyName) == false
+            && (apiKeyIds == null || apiKeyIds.length == 0)) {
+            return;
+        }
```

Comment:
Instead of skipping the test, we can just re-random one of the variables to be not null, e.g. replace `return;` with `username = randomAlphaOfLengthBetween(3, 8);`